src/transform/transform.ml

```diff
@@ -1,16 +1,42 @@
 open Prelude
 open GoblintCil
+module M = Messages
+
+(* The GADT-based approach of Query.t is overkill here *)
+type queries = {
+  ask : ?node:Node.t -> Cil.location -> Queries.ask ;
+  must_be_dead : stmt -> bool ;
+  must_be_uncalled : fundec -> bool ;
+}
 
 module type S = sig
-  val transform : (?node:Node.t -> Cil.location -> Queries.ask) -> file -> unit (* modifications are done in-place by CIL :( *)
+  val transform : queries -> file -> unit (* modifications are done in-place by CIL :( *)
+  val name : string
+  val requires_file_output : bool
 end
 
 let h = Hashtbl.create 13
-let register name (module T : S) = Hashtbl.add h name (module T : S)
-let run name =
-  let module T = (val try Hashtbl.find h name with Not_found -> failwith @@ "Transformation "^name^" does not exist!") in
-  if GobConfig.get_bool "dbg.verbose" then print_endline @@ "Starting transformation " ^ name;
-  T.transform
+let register (module T : S) = Hashtbl.replace h T.name (module T : S)
+
+let run_transformations ?(file_output = true) file names ask =
+  let active_transformations =
+    List.filter_map
+    (fun name ->
+      match Hashtbl.find_option h name with
+      | Some t -> Some (name, t)
+      | None -> M.warn_noloc "Transformation %s does not exist!" name; None)
```

Comment:
Updated to a hard failure. In a normal Goblint run, a hard failure also occurs on option parsing because of the enumeration of available transformations:

https://github.com/goblint/analyzer/blob/8d862841004ebe7800bea66de0da56e45e074b88/src/util/options.schema.json#L1360-L1363